### PR TITLE
add SSL support in NewServer

### PR DESCRIPTION
**Pull request by DanielGrossman**

### CODE REVIEW

File: smpp/server.go [hunk 1]
```diff
@@ -127,6 +125,21 @@ func NewUnstartedServer(user, password string, listener net.Listener) Server {
...
+		Passwd: password,
+		TLS:    tlsConf,
+		m:      map[pdu.ID]RequestHandlerFunc{},
+		a:      map[pdu.ID]AuthRequestHandlerFunc{},
+		s:      map[string]Session{},
+		l:      listener,
+	}
+	return s
+}
+
```
Comment (cesarrodrig) [inline smpp/server.go:142]: Where is the TLS field used in server? I think you still need to use it somewhere in the connection but I am not sure where.

Comment (DanielGrossman) [reply smpp/server.go:142]: Just realised the whole server bit of our go-smpp was copy-pasted from the fiorix test. I assumed that if server had a tls field it's probably implemented and should just be populated, apparently it's not :(. Not sure what their purpose was then. I'll look into it now.

Comment (cesarrodrig) [reply smpp/server.go:142]: Yeah, this server started with fiorix as just a mock Server and we have been modifying to our needs into a more production server. You can look at `conn.go`, I think TLS is used there for the client.

Comment (DanielGrossman) [reply smpp/server.go:142]: thanks, I think I found a better way to do this without actually modifying go-smpp, will ping you to have another look at the esme PR, hopefully we could scrap this PR.

Comment (DanielGrossman) [reply smpp/server.go:142]: @cesar0094 [here](https://github.com/veoo/spice/pull/190) is the PR, I think we can close this branch.

**Pull request closed without merging**